node/https.ts

```diff
@@ -16,12 +18,195 @@ export class Server {
 export function createServer() {
   notImplemented();
 }
-export function get() {
-  notImplemented();
+
+interface RequestOptions {
+  agent?: Agent | boolean;
+  auth?: string;
+  // deno-lint-ignore ban-types
+  createConnection?: Function;
+  defaultPort?: number;
+  family?: number;
+  // deno-lint-ignore ban-types
+  headers?: Object;
+  hints?: number;
+  host?: string;
+  hostname?: string;
+  insecureHTTPParser?: boolean;
+  localAddress?: string;
+  localPort?: number;
+  // deno-lint-ignore ban-types
+  lookup?: Function;
+  maxHeaderSize?: number;
+  method?: string;
+  path?: string;
+  port?: number;
+  protocol?: string;
+  setHost?: boolean;
+  socketPath?: string;
+  timeout?: number;
+  signal?: AbortSignal;
+  href?: string;
+}
+
+// Store additional root CAs.
+// undefined means NODE_EXTRA_CA_CERTS is not checked yet.
+// null means there's no additional root CAs.
+let caCerts: string[] | undefined | null;
+
+/** Makes a request to an https server. */
+export function get(
+  url: string | URL,
+  cb?: (res: HttpsIncomingMessage) => void,
+): HttpsClientRequest;
+export function get(
+  opts: RequestOptions,
+  cb?: (res: HttpsIncomingMessage) => void,
+): HttpsClientRequest;
+export function get(
+  url: string | URL,
+  opts: RequestOptions,
+  cb?: (res: HttpsIncomingMessage) => void,
+): HttpsClientRequest;
+// deno-lint-ignore no-explicit-any
+export function get(...args: any[]) {
+  const req = request(args[0], args[1], args[2]);
+  req.end();
+  return req;
 }
+
 export const globalAgent = undefined;
-export function request() {
-  notImplemented();
+/** HttpsClientRequest class loosely follows http.ClientRequest class API. */
```

Comment:
Shouldn't this class match `http.ClientRequest` exactly?

Reply:
Yup, this should be as close as possible to http.ClientRequest. The comment describes the current state. I'll continue work on that on follow-up PRs.

By the way, there're some props which don't make sense with this `fetch` implementation. For example, `req.socket` is a Duplex stream of underlying tcp connection, which we probably can't provide in our code.